### PR TITLE
Bigchanges

### PR DESCRIPTION
**Pull request by mcroitor**

### CODE REVIEW

File: samples/gameSample/headers/tui.hpp [hunk 1]
```diff
@@ -5,10 +5,51 @@
 #include <vector>
 #include <string>
 
+/**
+ * @brief class for communication with I/O
+ * 
+ */
 struct TUI {
```
Comment (AntonC9018) [inline samples/gameSample/headers/tui.hpp:12]: Why not a namespace + free funcs?

File: samples/gameSample/src/tui.cpp [hunk 2]
```diff
@@ -41,5 +41,30 @@ void TUI::Draw(const std::vector<std::string> &data)
...
+void TUI::Clear(Point position, size_t count) {
+    std::string blank(count, ' ');
+    TUI::Write(position, blank);
+}
+
+void TUI::ClearScreen() {
+    CONSOLE_SCREEN_BUFFER_INFO csbi;
+    size_t columns, rows;
+  
+    GetConsoleScreenBufferInfo(GetStdHandle(STD_OUTPUT_HANDLE), &csbi);
```
Comment (AntonC9018) [inline samples/gameSample/src/tui.cpp:61]: Oh nice win api

File: samples/gameSample/src/tui.cpp [hunk 3]
```diff
@@ -29,9 +29,9 @@ int TUI::scanKeys()
     return VK_NONAME;
 }
 
-void TUI::setPosition(point_t position)
+void TUI::SetCursor(Point position)
 {
-    std::cout << "\033[" << position.x << ";" << position.y << "H";
+    std::cout << "\033[" << position.y << ";" << position.x << "H";
```
Comment (AntonC9018) [inline samples/gameSample/src/tui.cpp:34]: https://learn.microsoft.com/en-us/windows/console/setconsolecursorposition
They seem to have APIs for this as well.

File: samples/gameSample/headers/tui.hpp (hunk 1 above)
Comment (mcroitor) [reply samples/gameSample/headers/tui.hpp:12]: Plan to introduce some private static variables, for screen size keeping, background color, text color etc.

Comment (AntonC9018) [reply samples/gameSample/headers/tui.hpp:12]: Well you can just add those as static functions in the file with the implementation. You don't necessarily have to declare your privates in the header at all.

Comment (mcroitor) [reply samples/gameSample/headers/tui.hpp:12]: I can`t. I talk about data restoration after changes, I need to keep data outside of functions. What is a problem with a static classes you see? It is just a nice example of singleton.

Comment (AntonC9018) [reply samples/gameSample/headers/tui.hpp:12]: I don't know what that means.
I don't really have a problem, I just don't see any benefit and I just think about this differently I guess (encapsulation on a module level).

Comment (AntonC9018) [reply samples/gameSample/headers/tui.hpp:12]: > Plan to introduce some private static variables, for screen size keeping, background color, text color etc.

Ah sorry I read this as "private static functions". Yeah keeping state there makes sense

**Pull request merged**